matchbox_socket/src/webrtc_socket/native.rs

```diff
@@ -82,127 +83,222 @@ impl Signaller for NativeSignaller {
+        config: &WebRtcSocketConfig,
+    ) -> (PeerId, Vec<Self::DataChannel>, Self::PeerLoopArgs) {
+        let (to_peer_message_tx, to_peer_message_rx) = new_senders_and_receivers(config);
+
+        debug!("handshake_accept");
+        let (connection, trickle) =
+            create_rtc_peer_connection(signal_peer.clone(), peer_state_tx.clone(), config)
+                .compat()
+                .await
+                .unwrap();
+
+        let (channel_ready_tx, wait_for_channels) = create_data_channels_ready_fut(config);
+        let data_channels = create_data_channels(
+            &connection,
+            channel_ready_tx,
+            signal_peer.id.clone(),
+            peer_state_tx.clone(),
+            messages_from_peers_tx,
+            &config.channels,
+        )
+        .await;
+
+        let offer = loop {
+            match from_peer_rx.next().await.expect("error") {
+                PeerSignal::Offer(offer) => {
+                    break offer;
+                }
+                _ => {
+                    warn!("ignoring other signal!!!");
+                }
             }
+        };
+        debug!("received offer");
+        let remote_description = RTCSessionDescription::offer(offer).unwrap();
+        connection
+            .set_remote_description(remote_description)
+            .await
+            .unwrap();
 
-            // TODO: maybe use some forward trait instead?
-            message = next_peer_messages_out.next().fuse() => {
-                match message {
-                    Some((channel_index, Some((peer, packet)))) => {
-                        let senders = connected_peers.get_mut(&peer)
-                            .unwrap_or_else(|| panic!("couldn't find data channel for peer {peer}"));
-                        let sender = senders.get_mut(channel_index)
-                            .unwrap_or_else(|| panic!("Unexpected data channel index during send: {channel_index}"));
-                        sender.unbounded_send(packet).unwrap();
-                    },
-                    Some((_, None)) | None => {
-                        // Receiver end of outgoing message channel closed,
-                        // which most likely means the socket was dropped.
-                        // There could probably be cleaner ways to handle this,
-                        // but for now, just exit.
-                        debug!("Outgoing message queue closed");
-                        break;
+        let answer = connection.create_answer(None).compat().await.unwrap();
+        signal_peer.send(PeerSignal::Answer(answer.sdp.clone()));
+        connection
+            .set_local_description(answer)
+            .compat()
+            .await
+            .unwrap();
+
+        let trickle_fut =
+            complete_handshake(trickle, &connection, from_peer_rx, wait_for_channels).await;
+
+        (
+            signal_peer.id,
+            to_peer_message_tx,
```

Comment:
Took me a while to understand that this is the `Self::DataChannels` arg. Maybe we should think about renaming the trait?

`PeerDataSender`? idk